### DISCUSSION TITLE
Enable Copy-PnPFile / Move-PnpFile to transfer between Odfb and SPO

### DISCUSSION THREAD

**Discussion by colonelclaypoo** [pnp/powershell#620] [Ideas] [answered]
I was hoping that I could use Copy-PnPFile to copy a file from OneDrive for Business to SharePoint Online. Unfortunately I can't get it to work. I suspect that being connected to only one root, i.e. https://[tenant].sharepoint.com/ or the mySite https://[tenant]-my.sharepoint.com/ is the issue. One would have to establish a connection to the target. Is this something that could be useful? Or is there already a way?

Thanks.

**Comment by omarmallat** [comment]
having the same case, where I need to do the opposite: from SPO to ODFB... the functionality is available on the web version so I expect it should be available

**Reply by colonelclaypoo** [reply]
@omarmallat, what do you mean with web version?

**Reply by omarmallat** [reply]
if you open onedrive website, you can select any folder and press "Move To". it gives you option to move to SharePoint sites (if the same account has access to these sites). it works well if you have a limited number of files/folders to move. but if you have lots of folders it won't work as it has limitations related to the size of the folders, which is my case.
I was looking for powershell to override that limitation by going through sub folders and move them instead of moving the parent folder which is too big.

**Comment by LeonArmston** [comment, marked as answer]
I dont think this is currently possible with Copy-PnPFile or Move-PnP file and I suspect this is due to authentication.

The way I've had to do it in the past which is a little clunky but works (see example below you could adapt). 

Is connecting to the OneDrive site, assigning the file to a Stream variable and then connecting to the SP Site and then using it with Add-PnPfile.

```powershell
#Connect To OneDrive
Connect-PnPOnline -Url 'https://tenantname-my.sharepoint.com/personal/username'

#Add the file to Stream Variable
$streamResult = (Get-PnPFile -Url '/personal/username/Documents/Certs/ReportsApp.cer').OpenBinaryStream()
Invoke-PnPQuery # this is will load the clientresult object returned from OpenBinaryStream

#Connect to SP Site
Connect-PnPOnline -Url 'https://tenantname.sharepoint.com/sites/YourSPSite'
Add-PnPFile -FileName 'ReportsApp.cer' -Folder "Shared Documents" -Stream $streamResult.Value
```
